source/common/network/utility.cc

```diff
@@ -243,18 +243,30 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
 }
 
 bool Utility::isInternalAddress(const char* address) {
+  // First try as an IPv4 address.
   in_addr addr;
   int rc = inet_pton(AF_INET, address, &addr);
-  if (1 != rc) {
-    return false;
+  if (rc == 1) {
+    // Handle the RFC1918 space for IPV4. Also count loopback as internal.
+    uint8_t* address_bytes = reinterpret_cast<uint8_t*>(&addr.s_addr);
+    if ((address_bytes[0] == 10) || (address_bytes[0] == 192 && address_bytes[1] == 168) ||
+        (address_bytes[0] == 172 && address_bytes[1] >= 16 && address_bytes[1] <= 31) ||
+        addr.s_addr == htonl(INADDR_LOOPBACK)) {
+      return true;
+    }
```

Comment:
can we return false in this case, to prevent trying again on IPv6. We will get better perf this way in the IPv4 case.